src/main/webapp/WEB-INF/jsp/fragments/common/header.jsp

```diff
@@ -60,7 +60,7 @@
                         </a>
                         <div class="dropdown-menu" aria-labelledby="navbarDropdown">
                             <c:forEach items="${category.childCategories}" var="childCategory">
-                                <a class="dropdown-item" href="#">${childCategory.name}</a>
+                                <a class="dropdown-item" href="productList/${childCategory.id}">${childCategory.name}</a>
```

Comment:
Url is not correct. Use <c:url> tag. And url should start from / like /productList..